### PR TITLE
Add test for __wakeup and update shields

### PR DESCRIPTION
**Pull request by dzubchik**

### CODE REVIEW

File: test/ImmutableTraitTest.php [hunk 1]
```diff
@@ -90,4 +90,22 @@ public function it_returns_the_default_attribute_value_of_properties_that_have_n
     {
         $this->assertSame([], $foo->bazzes);
     }
+
+    /**
+     * @test
+     * @dataProvider objectProvider
+     */
+    public function is_should_have_the_same_values_adter_deserealization(Foo $foo)
```
Comment (jdreesen) [inline test/ImmutableTraitTest.php]: There are some small typos in the method name:

```diff
- is_should_have_the_same_values_adter_deserealization
+ it_should_have_the_same_values_after_deserealization
```

**Comment by dzubchik** [comment]
@jdreesen , thank you!

**Comment by matthiasnoback** [comment]
Okay, thanks. Just merged it and pushed a new tag.

There's one conceptual problem I'm thinking about now: you added a test, but no changes to the production code (of course, they had already been made). So, is your test proving anything about the `Immutable` trait that wasn't already true? Does it characterize anything about the trait that needs to be preserved as "test case knowledge". I guess it does but it's just nice to consider for a moment ;)

**Comment by dzubchik** [comment]
Yes, you are right. We have created private fork, test it and use it in our code base. After you have merged previous PR we changed dependencies to use this repository. This test just need to illustrate problem that we faced in project. Thank you, for merging!

**Comment by matthiasnoback** [comment]
Excellent point, thanks :)

**Pull request merged**